### PR TITLE
Adding `AdditiveGroup` and `PrimeGroup`

### PR DESCRIPTION
**Pull request by autquis**
See arkworks-rs/algebra#577.

I changed `Cargo.toml` to avoid redoing what arkworks-rs/r1cs-std#122 has done.

closes: #XXXX

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (master)
- [ ] Linked to Github issue with discussion and accepted design OR have an explanation in the PR that describes this work.
- [ ] Wrote unit tests
- [ ] Updated relevant documentation in the code
- [ ] Added a relevant changelog entry to the `Pending` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

### CODE REVIEW

**Comment by mmagician** [comment]
I would prefer we first merge the r1cs-std PR to not have master here depend on any forks.

**Comment by mmaker** [comment]
r1cs-std's PR has been merged!

**Comment by weikengchen** [comment]
Let me update this PR to see if we can get it ready...

File: Cargo.toml [hunk 1]
```diff
@@ -71,4 +71,4 @@ ark-ec = { git = "https://github.com/arkworks-rs/algebra/" }
 ark-poly = { git = "https://github.com/arkworks-rs/algebra/" }
 ark-serialize = { git = "https://github.com/arkworks-rs/algebra/" }
 ark-algebra-test-templates = { git = "https://github.com/arkworks-rs/algebra/" }
-ark-r1cs-std = { git = "https://github.com/arkworks-rs/r1cs-std/" }
+ark-r1cs-std = { git = "https://github.com/mmaker/ark-r1cs-std/", branch = "feature/additive-groups" }
```
Comment (weikengchen) [inline Cargo.toml:74]:
```suggestion
ark-r1cs-std = { git = "https://github.com/arkworks-rs/r1cs-std/" }
```

**Review by weikengchen — APPROVED** [review]

**Comment by mmaker** [comment]
any news on this?

**Comment by mmaker** [comment]
ping.
this is causing problems for the folks at https://github.com/zkCollective/ and I need it for a presentation at DEF CON.

**Pull request merged**